### PR TITLE
Custom getter */ setter for ref / reactive

### PR DESCRIPTION
**Pull request by wellrus**
[Rendered](https://github.com/wellrus/rfcs/blob/master/active-rfcs/0000-custom-getter-setter-on-ref-and-reactive.md)

### CODE REVIEW

**Comment by ycmjason** [comment]
Getters and setters are still available in `computed` if I am not wrong. So there is no need to make ref and reactive too overly complicated.

**Comment by jacekkarczmarczyk** [comment]
But you need additional data to store your value

**Comment by LinusBorg** [comment]
I don't see an issue with that. or to put it another way, if you want to keep this vallue internal, create a one-line wrapper around `computed`.

**Comment by posva** [comment]
As mentioned this over complicates the api with no gain (memory usage won't be different).
On top of that this can be implemented locally:

```js
function computedRef(value, { get, set }) {
  const refValue = ref(value)
  return computed({
  // some wrapping of get and set
 })
}
```

**Comment by ycmjason** [comment]
@posva

I believe that there isn't a way to have deep getter/setter tho? Would it be useful if computed provides a way to get/set deeply? Currently the getter and setter will only be triggered at `.value` level?

**Comment by posva** [comment]
@ycmjason  what do you mean by having a deep getter/setter? How is that related to this RFC?

**Comment by ycmjason** [comment]
It is related to this RFC since what's mentioned in the RFC could not be fully implemented if computed can only return Ref.

Let me rephrase my previous comment. It wasn't very clear. Perhaps it would be useful to allow computed to return an reactive object where setting/getting of its entries will trigger the getter/setter? Much like how proxy traps works?

**Comment by ycmjason** [comment]
Although conceptually that's the same as an reactive with many computed values, but that wouldn't allow new properties to be added to the reactive?

**Comment by yyx990803** [comment]
Technically this is indeed something that the current API does not support, however what I find not convincing enough is the Motivation section. It is too brief, and "may be useful" simply doesn't justify feature additions like this. I would like to see what **real world use cases** this can solve or make simpler.

**Comment by donnysim** [comment]
The only real use case I could see is having value validator as in:

```js
price: {
  get() {
    return this._price;
  },
  set(value) {
    if (isNaN(Number(value))) {
      this._price = 0;
    } else {
      this._price = Number(value);
    }
  },
},
```
or global state getter/mutator without having to create a bunch of computed properties, but if what @posva said would work the same then I don't really see the need of it other than convenience. Maybe just the thought of computed property having a setter sounds weird, as it's more often associated with just retrieving a value (very few knew they can have a setter on a computed from my encounters/helping others).
